### ISSUE TITLE
Silent renew failing due to required session_state

### ISSUE THREAD

**Issue by mikecowgill** [damienbod/angular-auth-oidc-client#434]
I'm not able to get silent renew working with my oidc server using code flow. After some debugging it looks like the session_state is required to start the token request in the silentRenewEventHandler:

```
private silentRenewEventHandler(e: CustomEvent) {
        this.loggerService.logDebug('silentRenewEventHandler');

        if (this.authConfiguration.response_type === 'code') {

            const urlParts = e.detail.toString().split('?');
            const params = new HttpParams({
                fromString: urlParts[1]
            });
            const code = params.get('code');
            const state = params.get('state');
            const session_state = params.get('session_state');
            const error = params.get('error');
```
```
            if (code && state && session_state) {
                this.requestTokensWithCodeProcedure(code, state, session_state);
            }
```

My oidc server is not sending the session_state parameter and so the requestTokensWithCodeProcedure is never called and it continues to loop.

Similar to #371 should this be an optional parameter?

**Comment by damienbod** [comment]
@mikecowgill 

Thanks for reporting. I'll fix this tomorrow and do a new version

https://github.com/damienbod/angular-auth-oidc-client/blob/master/projects/angular-auth-oidc-client/src/lib/services/oidc.security.service.ts#L893

Greetings Damien

**Comment by damienbod** [comment]
@mikecowgill fixed in version 9.0.6

Thanks for reporting

**Comment by mikecowgill** [comment]
@damienbod Wow thanks for the quick response and fix!
